CPS-????/README.md

```diff
@@ -0,0 +1,81 @@
+
+Any time a developer creates a transaction level spending script, it will be redundantly executed for every additional UTxO consumed from the corresponding script address. These redundant executions are a waste of scarce resources and result in much higher fees for end users. Not only that, but the inability to efficiently use these transaction level spending scripts significantly handicaps the use of the eUTxO model. The eUTxO model already allows for creating such spending scripts; developers are just unable to properly use them.
+
+### The Current Alternative
+The alternative involves taking advantage of the fact that staking scripts do not suffer from the same limitation as spending scripts.
+
+> Given a [spending] script **s**, you can turn it into a [spending] script **s'** that checks that 0 ADA is withdrawn from a [staking] script **s''** with the same logic as **s**. This is a transformation that is valid for any [spending] script.    -- @L-as
+
+In a nut shell, you use a staking script **AS** a spending script. The transaction level logic would be in the staking script. Meanwhile, the spending script just checks that both the required staking script is executed AND 0 ADA is withdrawn from the rewards address within the transaction. The withdrawal of the 0 ADA is necessary to force the staking script to be executed.
+
+The only requirement to use this technique is that the script address must have been created using the `s'` spending script and the `s''` staking script. The staking address DOES NOT need to actually have rewards or even be delegated/registered for this technique to work.
+
+With this method, the main computation is only executed once no matter how many UTxOs are spent from the script address.
+
+#### The Drawbacks
+
+1. This technique now means that everything requires two plutus scripts instead of just one. While the spending script can be made extremely small, this doesn't change the fact that two are now needed.
+2. This technique DOES NOT stop the redundant executions; the spending script will still redundantly check if the required staking script was executed. This technique just minimizes the cost of the redundant executions by moving the heavy computation into a script that is only executed once per transaction.
+3. This is not the intended use for staking scripts.
+
+### Simple Scripts
+I tried to test Cardano simple scripts to determine if this problem also impacts them. However, given that simple scripts are so small, it was hard to tell from just looking at the transaction fee. Since the interface for using Cardano simple scripts is the same as for using plutus scripts, I have to assume that this problem also impacts simple scripts. 
+
+If simple scripts are indeed impacted by this problem, the above technique would not work for them. The technique only works for plutus scripts.
+
+## Use Cases
```

Comment:
I'd like to see a lot more detail here, of the level that there is for the swap example. At the moment it looks to me like there is _one_ convincing usecase, which isn't much.